arangosh/Benchmark/testcases/RandomShapesTest.h

```diff
@@ -25,98 +25,78 @@
-      TRI_AppendString2StringBuffer(buffer, key.c_str(), key.size());
-      TRI_AppendStringStringBuffer(buffer, "\"");
-
-      uint32_t const t = _randomValue % (uint32_t)(globalCounter + threadNumber + 1);
-
-      for (uint64_t i = 1; i <= n; ++i) {
-        TRI_AppendStringStringBuffer(buffer, ",\"value");
-        TRI_AppendUInt64StringBuffer(buffer, (uint64_t)(globalCounter + i));
-        if (t % 3 == 0) {
-          TRI_AppendStringStringBuffer(buffer, "\":true");
-        } else if (t % 3 == 1) {
-          TRI_AppendStringStringBuffer(buffer, "\":null");
-        } else
-          TRI_AppendStringStringBuffer(
-              buffer,
-              "\":\"some bogus string value to fill up the datafile...\"");
+    void tearDown() override {}
+
+    std::string url(int const threadNumber, size_t const threadCounter,
+                    size_t const globalCounter) override {
+      size_t const mod = globalCounter % 3;
+      if (mod == 0) {
+        return std::string("/_api/document?collection=") + _arangobench.collection();
+      } else {
+        size_t keyId = (size_t)(globalCounter / 3);
+        std::string const key = "testkey" + StringUtils::itoa(keyId);
+        return std::string("/_api/document/") + _arangobench.collection() +
+          std::string("/") + key;
       }
+    }
 
-      TRI_AppendStringStringBuffer(buffer, "}");
-
-      *length = TRI_LengthStringBuffer(buffer);
-      *mustFree = true;
-      char* ptr = TRI_StealStringBuffer(buffer);
-      TRI_FreeStringBuffer(buffer);
+    rest::RequestType type(int const threadNumber, size_t const threadCounter,
+                           size_t const globalCounter) override {
+      size_t const mod = globalCounter % 3;
+      if (mod == 0) {
+        return rest::RequestType::POST;
+      } else if (mod == 1) {
+        return rest::RequestType::GET;
+      } else {
+        return rest::RequestType::DELETE_REQ;
+      }
+    }
 
-      return (char const*)ptr;
-    } else {
-      *length = 0;
-      *mustFree = false;
-      return (char const*)nullptr;
+    void payload(int threadNumber, size_t threadCounter,
+                 size_t globalCounter, std::string& buffer) const override {
+      size_t mod = globalCounter % 3;
+      if (mod == 0) {
+        uint64_t n = _arangobench.complexity();
+        size_t keyId = static_cast<size_t>(globalCounter / 3);
+        using namespace arangodb::velocypack;
+        Builder b;
+        b.openObject();
+        b.add("_key", Value(std::string("testkey") + std::to_string(keyId)));
+        uint32_t t = _randomValue % static_cast<uint32_t>(globalCounter + threadNumber + 1);
+        for (uint64_t i = 1; i <= n; ++i) {
+          std::string fieldName = "value" + std::to_string(static_cast<uint64_t>(globalCounter + i));
+          if (t % 3 == 0) {
+            b.add(fieldName, Value(true));
+          } else if (t % 3 == 1) {
+            b.add(fieldName, Value(NULL));
```

Comment:
IMO `NULL` is equivalent to a numeric value of `0`, so it will be stored as a number. We want a JSON `null` here instead:
```suggestion
            b.add(fieldName, Value(ValueType::Null));
```